SwiftWisdom/Core/StandardLibrary/Numbers/Mathable.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Mathable.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/12/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol Mathable: Comparable {
+    func + (lhs: Self, rhs: Self) -> Self
+    func / (lhs: Self, rhs: Self) -> Self
+    func * (lhs: Self, rhs: Self) -> Self
+    func - (lhs: Self, rhs: Self) -> Self
+    init(_ int: IntMax)
+}
+
+extension Mathable {
+    public static var zero: Self {
+        return self.init(0.toIntMax())
+    }
+}
+
+extension CollectionType where Generator.Element: Mathable {
+    
+    public func ip_average() -> Generator.Element {
+        return ip_mean()
+    }
+    
+    public func ip_mean() -> Generator.Element {
+        let zero = Generator.Element.zero
+        let count = Generator.Element(self.count.toIntMax())
+        guard count > zero else { return zero }
+        return reduce(zero, combine: +) / count
+    }
+    
+    public func ip_median() -> Generator.Element {
+        let zero = Generator.Element.zero
+        let count = Generator.Element(self.count.toIntMax())
+        guard count > zero else { return zero }
+        return sort(<).ip_middleValues.ip_mean()
+    }
+    
+    public func ip_mode() -> [Generator.Element] {
```

Comment:
I like the symmetry of having mean, median, and mode for these mathable types. And then if your array / collection isn't mathable then you can still find the `ip_mostCommonElements`. It would feel incomplete with just mean and median.